library/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java

```diff
@@ -163,7 +166,9 @@ public ArrayList<OCShare> parseXMLResponse(InputStream is) throws XmlPullParserE
 			parser.setInput(is, null);
 			parser.nextTag();
 			return readOCS(parser);
-
+		} catch (Exception e) {
+			Log_OC.e(this, "Error parsing xml", e);
+			return new ArrayList<>();
```

Comment:
Judging from the method signature, `parseXMLResponse` is supposed to throw exceptions, not catch them and return an empty list.

Reply:
Totally true!